packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,103 @@
+import { Account, Address, toBuffer, toType, TypeOutput } from '@ethereumjs/util'
+import { BaseStateManager, StateManager } from '.'
+import { StorageDump } from './interface'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+import { keccak256 } from 'ethereum-cryptography/keccak'
+import { Common } from '@ethereumjs/common'
+import { Cache } from './cache'
+
+export interface EthersStateManagerOpts {
+  common?: Common
+  provider: JsonRpcProvider
+}
+
+export class EthersStateManager extends BaseStateManager implements StateManager {
+  private provider: JsonRpcProvider
+  private contractCache: Map<string, Buffer>
+  private storageCache: Map<string, string>
+
+  constructor(opts: EthersStateManagerOpts) {
+    super({ common: opts.common })
+    this.provider = opts.provider
+    this.contractCache = new Map()
+    this.storageCache = new Map()
```

Comment:
should we do this for basestatemanager downstream?

then the task will only come down to just cache load/flush.

Reply:
I haven't thought that deeply on it yet.  I need to look more into the underlying functionality of the `baseStateManager` class versus our existing `DefaultStateManager` and see how it works.  The reason I'm storing them in a cache in this instance is because this state manager is "readonly" and we don't persist results that occur through VM execution beyond the immediate context of the code this is running in and I didn't want any contracts created/storage slots changed to get lost in the immediate context since the `DefaultStateManager` is storing all the updates directly to the various tries.